Submitted By: Michael Labuschke (michael at labuschke dot de)
Date: 2004-05-13
Initial Package Version: 1.3.29 
Origin: me
Upstream Status: not send
Description: Fixes head -1 and tail -1 issues. 
 
diff -Naur apache_1.3.31.orig/src/Configure apache_1.3.31/src/Configure
--- apache_1.3.31.orig/src/Configure	2004-04-09 17:01:50.000000000 +0000
+++ apache_1.3.31/src/Configure	2004-05-13 12:47:30.861911755 +0000
@@ -76,7 +76,7 @@
 ## Now see if Configuration.tmpl is more recent than $file. If
 ## so, then we complain and bail out
 ##
-if ls -lt $file Configuration.tmpl | head -1 | \
+if ls -lt $file Configuration.tmpl | head -n 1 | \
   grep 'Configuration.tmpl' > /dev/null
 then
   echo "Configuration.tmpl is more recent than $file;"
@@ -897,7 +897,7 @@
 #
 # If $TCC is null, then no such line exists in Configuration
 #
-TCC=`egrep '^CC=' Makefile.config | tail -1 | awk -F= '{print $2}'`
+TCC=`egrep '^CC=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
 if [ "x$TCC" = "x" ]; then
     if [ "x$CC" = "x" ]; then
 	# At this point, CC is not set in Configuration or above, so we
@@ -937,7 +937,7 @@
 ####################################################################
 ## Now check how we can _directly_ run the C pre-processor
 ##
-TCPP=`egrep '^CPP=' Makefile.config | tail -1 | awk -F= '{print $2}'`
+TCPP=`egrep '^CPP=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
 if [ "x$TCPP" != "x" ]; then
     CPP=`CPP="$TCPP"; export CPP CC; ./helpers/findcpp.sh`
 else
@@ -1004,18 +1004,18 @@
 ####################################################################
 ## Look for OPTIM and save for later
 ##
-TOPTIM=`egrep '^OPTIM=' Makefile.config | tail -1 | awk -F= '{print $2}'`
-TRANLIB=`egrep '^RANLIB=' Makefile.config | tail -1 | awk -F= '{print $2}'`
-TTARGET=`egrep '^TARGET=' Makefile.config | tail -1 | awk -F= '{print $2}'`
+TOPTIM=`egrep '^OPTIM=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
+TRANLIB=`egrep '^RANLIB=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
+TTARGET=`egrep '^TARGET=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
 
 ####################################################################
 ## Check for user provided flags for shared object support
 ##
-TLD_SHLIB=`egrep '^LD_SHLIB=' Makefile.config | tail -1 | awk -F= '{print $2}'`
-TLDFLAGS_SHLIB=`egrep '^LDFLAGS_SHLIB=' Makefile.config | tail -1 | awk -F= '{print $2}'`
-TLDFLAGS_MOD_SHLIB=`egrep '^LDFLAGS_MOD_SHLIB=' Makefile.config | tail -1 | awk -F= '{print $2}'`
-TLDFLAGS_SHLIB_EXPORT=`egrep '^LDFLAGS_SHLIB_EXPORT=' Makefile.config | tail -1 | awk -F= '{print $2}'`
-TCFLAGS_SHLIB=`egrep '^CFLAGS_SHLIB=' Makefile.config | tail -1 | awk -F= '{print $2}'`
+TLD_SHLIB=`egrep '^LD_SHLIB=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
+TLDFLAGS_SHLIB=`egrep '^LDFLAGS_SHLIB=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
+TLDFLAGS_MOD_SHLIB=`egrep '^LDFLAGS_MOD_SHLIB=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
+TLDFLAGS_SHLIB_EXPORT=`egrep '^LDFLAGS_SHLIB_EXPORT=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
+TCFLAGS_SHLIB=`egrep '^CFLAGS_SHLIB=' Makefile.config | tail -n 1 | awk -F= '{print $2}'`
 
 ####################################################################
 ## Handle TARGET name
@@ -1796,7 +1796,7 @@
 	    if [ -f $modbase.c ] ; then
 		# Guess module structure name in case there is no
 		# module definition in this file
-		modname=`egrep '^module .*;' $modbase.c | head -1 |\
+		modname=`egrep '^module .*;' $modbase.c | head -n 1 |\
 			sed 's/^module.*[ 	][ 	]*//' | \
 			sed 's/[ 	]*;[ 	]*$//'`
 		# Get any module definition part
@@ -2326,7 +2326,7 @@
 ## ap_config_auto.h so they are available to external modules needing to
 ## include Apache header files.
 ##
-TEXTRA_CFLAGS=`egrep '^EXTRA_CFLAGS=' Makefile.config | tail -1 |\
+TEXTRA_CFLAGS=`egrep '^EXTRA_CFLAGS=' Makefile.config | tail -n 1 |\
 	       sed -e 's;^EXTRA_CFLAGS=;;' -e 's;\`.*\`;;'`
 tmpstr=`echo $CFLAGS $TEXTRA_CFLAGS |\
 	sed -e 's;[ 	]\([+-]\);!\1;g' -e 's/\([^\\\]\)"/\1/g' -e 's/\\\"/\"/g'`
diff -Naur apache_1.3.31.orig/src/helpers/buildinfo.sh apache_1.3.31/src/helpers/buildinfo.sh
--- apache_1.3.31.orig/src/helpers/buildinfo.sh	2004-02-16 22:23:09.000000000 +0000
+++ apache_1.3.31/src/helpers/buildinfo.sh	2004-05-13 12:47:30.862911449 +0000
@@ -100,13 +100,13 @@
 esac
 if [ "x$domainname" = "x" ]; then
     if [ -f /etc/resolv.conf ]; then
-        domainname="`egrep '^[ 	]*domain' /etc/resolv.conf | head -1 |\
+        domainname="`egrep '^[ 	]*domain' /etc/resolv.conf | head -n 1 |\
                      sed -e 's/.*domain//' \
                          -e 's/^[ 	]*//' -e 's/^ *//' -e 's/^	*//' \
                          -e 's/^\.//' -e 's/^/./' |\
                      awk '{ printf("%s", $1); }'`"
         if [ "x$domainname" = "x" ]; then
-            domainname="`egrep '^[ 	]*search' /etc/resolv.conf | head -1 |\
+            domainname="`egrep '^[ 	]*search' /etc/resolv.conf | head -n 1 |\
                          sed -e 's/.*search//' \
                              -e 's/^[ 	]*//' -e 's/^ *//' -e 's/^	*//' \
                              -e 's/ .*//' -e 's/	.*//' \
diff -Naur apache_1.3.31.orig/src/helpers/fmn.sh apache_1.3.31/src/helpers/fmn.sh
--- apache_1.3.31.orig/src/helpers/fmn.sh	2004-02-16 22:23:09.000000000 +0000
+++ apache_1.3.31/src/helpers/fmn.sh	2004-05-13 12:47:30.864910838 +0000
@@ -36,7 +36,7 @@
     cat $modbase.module >$tmpfile
 else
     if [ -f $modbase.c ] ; then
-        modname=`egrep '^module .*;' $modbase.c | head -1 |\
+        modname=`egrep '^module .*;' $modbase.c | head -n 1 |\
                 sed 's/^module.*[ 	][ 	]*//' | \
                 sed 's/[ 	]*;[ 	]*$//'`
         if grep "MODULE-DEFINITION-" $modbase.c >/dev/null; then
diff -Naur apache_1.3.31.orig/src/helpers/getuid.sh apache_1.3.31/src/helpers/getuid.sh
--- apache_1.3.31.orig/src/helpers/getuid.sh	2004-02-16 22:23:09.000000000 +0000
+++ apache_1.3.31/src/helpers/getuid.sh	2004-05-13 12:47:30.866910226 +0000
@@ -48,7 +48,7 @@
     AP_WAIPATH=`./src/helpers/PrintPath whoami`
     AP_LOGNAME=`$AP_WAIPATH`
 else
-    AP_LOGNAME=`who am i | tail -1 | sed -e 's/[ 	][ 	]*.*$//'`
+    AP_LOGNAME=`who am i | tail -n 1 | sed -e 's/[ 	][ 	]*.*$//'`
 fi
 
 #
